paddle/framework/tensor.h

```diff
@@ -0,0 +1,122 @@
+#pragma once
+
+#include <memory>
+#include <type_traits>
+#include <typeinfo>
+#include "paddle/framework/ddim.h"
+#include "paddle/framework/enforce.h"
+#include "paddle/platform/assert.h"
+#include "paddle/platform/place.h"
+
+namespace paddle {
+namespace framework {
+
+class Tensor {
+  using paddle::platform::Place;
+  using paddle::platform::get_place;
+
+ public:
+  explicit Tensor(DDim dims) : dims_(dims), place_(get_place()) {}
```

Comment:
Please be aware that I followed @qingqing01 's suggestion 3 days ago https://github.com/PaddlePaddle/Paddle/pull/2611#pullrequestreview-46458062 and removed `Tensor::Tensor`, `Tensor::place_`, `Tensor::dims_` in my PR https://github.com/PaddlePaddle/Paddle/pull/2611 so to make a concise and consistent syntax.